include/networkit/graph/Graph.hpp

```diff
@@ -1363,7 +1363,153 @@ class Graph final {
      */
     Graph(count n = 0, bool weighted = false, bool directed = false, bool edgesIndexed = false);
 
-    Graph(const Graph &G, bool weighted, bool directed, bool edgesIndexed = false);
+    template <class EdgeMerger = std::plus<edgeweight>>
+    Graph(const Graph &G, bool weighted, bool directed, bool edgesIndexed = false,
+          EdgeMerger edgeMerger = std::plus<edgeweight>())
+        : n(G.n), m(G.m), storedNumberOfSelfLoops(G.storedNumberOfSelfLoops), z(G.z),
+          omega(edgesIndexed ? G.omega : 0), t(G.t), weighted(weighted), directed(directed),
+          edgesIndexed(edgesIndexed), // edges are not indexed by default
+          exists(G.exists),
+
+          // let the following be empty for the start, we fill them later
+          inEdges(0), outEdges(0), inEdgeWeights(0), outEdgeWeights(0), inEdgeIds(0), outEdgeIds(0),
+
+          // empty node attribute map as last member for this graph
+          nodeAttributeMap(this) {
+
+        if (G.isDirected() == directed) {
+            // G.inEdges might be empty (if G is undirected), but
+            // that's fine
+            inEdges = G.inEdges;
+            outEdges = G.outEdges;
+
+            // copy weights if needed
+            if (weighted) {
+                if (G.isWeighted()) {
+                    // just copy from G, again either both graphs are directed or both are
+                    // undirected
+                    inEdgeWeights = G.inEdgeWeights;
+                    outEdgeWeights = G.outEdgeWeights;
+                } else {
+                    // G has no weights, set defaultEdgeWeight for all edges
+                    if (directed) {
+                        inEdgeWeights.resize(z);
+                        for (node u = 0; u < z; u++) {
+                            inEdgeWeights[u].resize(G.inEdges[u].size(), defaultEdgeWeight);
+                        }
+                    }
+
+                    outEdgeWeights.resize(z);
+                    for (node u = 0; u < z; u++) {
+                        outEdgeWeights[u].resize(outEdges[u].size(), defaultEdgeWeight);
+                    }
+                }
+            }
+            if (G.hasEdgeIds() && edgesIndexed) {
+                inEdgeIds = G.inEdgeIds;
+                outEdgeIds = G.outEdgeIds;
+            }
+        } else if (G.isDirected()) {
+            // G is directed, but we want an undirected graph
+            // so we need to combine the out and in stuff for every node
+            outEdges.resize(z);
+            if (weighted)
+                outEdgeWeights.resize(z);
+            if (G.hasEdgeIds() && edgesIndexed)
+                outEdgeIds.resize(z);
+#pragma omp parallel for
+            for (omp_index u = 0; u < static_cast<omp_index>(z); ++u) {
```

Comment:
Why do we use these loops and not `parallelForNodes`? If `G` has deleted nodes then we do useless iterations, which are automatically skipped by graph iterators.